hexl/ntt/fwd-ntt-avx512.cpp

```diff
@@ -366,16 +422,22 @@ void ForwardTransformToBitReverseAVX512(
     const uint64_t* W = &root_of_unity_powers[W_idx];
     const uint64_t* W_precon = &precon_root_of_unity_powers[W_idx];
 
-    FwdT8<BitShift, false>(operand, v_neg_modulus, v_twice_mod, t, 1, W,
-                           W_precon);
+    if (result != operand) {
+      FwdT8<BitShift, false>(result, operand, v_neg_modulus,
+                              v_twice_mod, t, 1, W, W_precon);
+    }
+    else {
+      FwdT8<BitShift, false>(result, v_neg_modulus, v_twice_mod, t, 1, W,
```

Comment:
Can we just call the out-of-place FwdT8 here? Perhaps we don't need the in-place FwdT8 at all.